### PR TITLE
134 static reports

### PR DESCRIPTION
**Pull request by MemorySounds**
Basic outline of the approach:

- gather historic report data (used - with some refactoring - the already existing report data aggregation)
- create a workbook (excel file) - now happens in the backend instead of the frontend
- save the file to the db
- defined a task for the logger that runs all of the above ^

Next steps - 
- setup frontend UI in the admin panel to periodically check if a new report is available
- allow the user to download any existing report straight from the db

### CODE REVIEW

**Review by haydnba — COMMENTED** [review]
Hey Robert this looks brilliant - I've not pulled the branch and tested locally though. I think as long as you are happy they were good to get it staged and released as there doesn't appear to be anything there likely to bring down a critical service. 

Just a few questions:

1. I think you've added some services and endpoints that are intended to be available for use in future once we've set up a long running task worker process? If that's the case maybe could you just add comments to that effect so no one struggles to understand what they are there for

2. This is completely non-blocking but originally I was thinking we would store the report output in the db as json just under a ´mixed' data type. This was so that we could maybe directly display the results of the report in the client UI rather than having to download (with option of downloading as excel sheet available). I think you should definitely keep it as is but wonder what your thoughts are on that other option... it might be we could just add this as an additional parameter on the model and provide ´json' as well as ´buffer'

3. I think Chloé would really love it if the reports could be emailed to her and other stakeholders at generation time - we'd just need to add this in to the ´generate_historic_report' task

File: server/routes/api/statistics.js [hunk 1]
```diff
@@ -19,4 +21,33 @@ router.get('/report', async (req, res) => {
   res.json(data);
 });
 
+// generate historic report
+router.get('/historic', async (req, res) => {
```
[the remark this answers is not in the document]
Comment (MemorySounds) [reply server/routes/api/statistics.js:25]: Sorry, this certainly is confusing - I made these temporarily whilst I was testing the method. I added a little button on the frontend and was calling this to simulate a call from the scheduled task manager (probably could have done that easier)

Technically, frontend should not be concerned with generating the historic report on demand for now, as that crashes the app.

[the remark this answers is not in the document]
Comment (haydnba) [reply server/routes/api/statistics.js:25]: Makes sense thanks Robert!

**Review by haydnba — APPROVED** [review]
💯

**Pull request merged**